src/recipes/ai-context/index.ts

```diff
@@ -12,30 +13,51 @@ import {
   getExistingContext,
   parseContextFile,
   writeFile,
-  FILE_NAME,
+  NTL_DEV_MCP_FILE_NAME,
   NETLIFY_PROVIDER,
+  getContextConsumers,
+  ConsumerConfig,
+  deleteFile,
 } from './context.js'
 
 export const description = 'Manage context files for AI tools'
 
-const presets = [
-  { name: 'Windsurf rules (.windsurf/rules/)', value: '.windsurf/rules' },
-  { name: 'Cursor rules (.cursor/rules/)', value: '.cursor/rules' },
-  { name: 'Custom location', value: '' },
-]
+// context consumers endpoints returns all supported IDEs and other consumers
+// that can be used to pull context files. It also includes a catchall consumer
+// for outlining all context that an unspecified consumer would handle.
+const allContextConsumers = (await getContextConsumers(version)).filter((consumer) => !consumer.hideFromCLI)
```

Comment:
I think `getContextConsumers` can throw, since we're not wrapping the `fetch` call in a `try/catch`, so this could lead to an uncaught rejection?

Reply:
Is this async call to the `getContextConsumers` made in the main scope to try and fetch the data as soon as possible, so by the time user gets to the promp, we already have the needed data?

Even so, I suggest moving it to the beginning of the `run` function. I don't think there is a huge gain between calling it from the main scope and calling it from within the `run` function which should be the first thing that executes when user runs the command. There is something about executing `fetch` in the main scope feels wrong to me but I can't explain what it is exactly :)